src/nyx/features/caliper_feret.cpp

```diff
@@ -25,47 +24,28 @@ void CaliperFeretFeature::calculate(LR& r)
 	if (r.has_bad_data())
 		return;
 
-	std::vector<double> allD;	// Diameters at 0-180 degrees rotation
-	calculate_imp(r.convHull_CH, allD);
+	std::vector<float> angles;
+	std::vector<double> ferets;
+	calculate_angled_caliper_measurements (r.convHull_CH, angles, ferets);
 
-	// Calculate statistics of diameters
-	auto s = ComputeCommonStatistics2(allD);
-	_min = (double)s.min;
-	_max = (double)s.max;
+	// Statistics of Feret diameters
+	auto s = ComputeCommonStatistics2 (ferets);
+	_min = s.min;
+	_max = s.max;
 	_mean = s.mean;
 	_median = s.median;
 	_stdev = s.stdev;
-	_mode = (double)s.mode;
+	_mode = s.mode;
 
-	// Calculate angles at min- and max- diameter
-	if (allD.size() > 0)
-	{
-		// Min and max
-		auto itr_min_d = std::min_element(allD.begin(), allD.end());
-		auto itr_max_d = std::max_element(allD.begin(), allD.end());
-		minFeretDiameter = *itr_min_d;
-		maxFeretDiameter = *itr_max_d;
-
-		// Angles
-		auto idxMin = std::distance(allD.begin(), itr_min_d);
-		minFeretAngle = (double)idxMin / 2.0;
-		auto idxMax = std::distance(allD.begin(), itr_max_d);
-		maxFeretAngle = (double)idxMax / 2.0;
-	}
-	else
-	{
-		// Degenerate case
-		minFeretDiameter =
-		maxFeretDiameter =
-		minFeretAngle =
-		maxFeretAngle = 0.0;
-	}
+	// Angles
+	size_t min_i = Nyxus::get_min_idx (ferets);
+	size_t max_i = Nyxus::get_max_idx (ferets);

```

Comment:
Another question: Now that you have reworked the calculation, can we just use `std::min_element` and `std::max_element` here or it might produce different result based on OS/HW types?